### PR TITLE
feat(Components): Create Breadcrumbs Component

### PR DESCRIPTION
**Pull request by calcerrada**
#189

## TYPE
- Feature

## Description
Create Breadcrumbs Component

### Documentation
- [x] Have explained API (Inputs, Outpus, Models, Functions, etc)
- [x] Have example running
- [x] Have example code (HTML, ts)
- [x] Have changelog.md updated

### Code
- [x] Have 3 spaces indentation
- [x] Pass ts lint Task
- [ ] Pass Sass lint Task
- [x] Have QA-tags

### Tests
- [x] Have 70% Tests coverage

### Other observations
- Add other information you consider important

### Post-review reminder
- Remember reviewer approve changes inside the Files Changed tab (Review changes dropdown)
- Squash commits
- Release version in case of breaking changes and notify of all front

### CODE REVIEW

File: src/st-breadcrumbs/st-breadcrumbs.component.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export class StBreadCrumbs {
+   @Input() options: string[];
+   @Input() qaTag: string;
+   @Output() changeOption: EventEmitter<number> = new EventEmitter<number>();
+
+  public generateCrumbs(): string[] {
+      if (this.options.length <= 6) {
+         return this.options;
+      } else {
+         return this.options.slice(0, 1).concat(['...']).concat(this.options.slice(-4));
```
Comment (pjpenast) [inline src/st-breadcrumbs/st-breadcrumbs.component.ts]: In this line the state is mutating please try to make staless components using Object.assign

File: src/st-breadcrumbs/st-breadcrumbs.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
+@import '../../node_modules/@stratio/egeo-ui-base/utils/typography';
+@import '../../node_modules/@stratio/egeo-ui-base/utils/colors';
+
+$font-family:egeo-get-font(body);
+$text-color: egeo-get-color(a2) !default;
+$text-color---hover:egeo-get-color(n0);
+$text-color---selected:egeo-get-color(n10);
+$bg-color---hover:egeo-get-color(a1);
+$bg-color---main-option:egeo-get-color(n1);
+$font-size: egeo-get-font-size(head_xxsmall) !default;
```
Comment (pjpenast) [inline src/st-breadcrumbs/st-breadcrumbs.scss]: The styles should be in egeo-theme.

File: src/modules.ts [hunk 3]
```diff
@@ -43,5 +44,6 @@ export const DECLARATIONS: any[] = [
    StVerticalTabsModule,
    StPaginationModule,
    StModalModule,
-   StRadioModule
+   StRadioModule,
+   StBreadcrumbsModule
```
Comment (ppenalver) [inline src/modules.ts]: Alphabetical order

File: src/st-breadcrumbs/st-breadcrumbs.component.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+@Component({
+   selector: 'st-breadcrumbs',
+   templateUrl: './st-breadcrumbs.html',
+   styleUrls: ['./st-breadcrumbs.scss']
+})
+
+export class StBreadCrumbs {
+   @Input() options: string[];
+   @Input() qaTag: string;
```
Comment (ppenalver) [inline src/st-breadcrumbs/st-breadcrumbs.component.ts]: check required

File: src/st-breadcrumbs/st-breadcrumbs.component.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+         return this.options.slice(0, 1).concat(['...']).concat(this.options.slice(-4));
+      }
+   }
+
+   public onSelect(index: number): void {
+
+      if (index + 1 < this.options.length) {
+         if (this.options.length <= 6 || index === 0) {
+            this.changeOption.emit(index);
+         }else {
```
Comment (ppenalver) [inline src/st-breadcrumbs/st-breadcrumbs.component.ts]: tslint space between else and }

File: src/st-breadcrumbs/st-breadcrumbs.module.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+@NgModule({
+   imports: [CommonModule],
+   exports: [StBreadCrumbs],
+   declarations: [StBreadCrumbs],
+   entryComponents: [StBreadCrumbs],
+   providers: []
+})
+export class StBreadcrumbsModule {
+   static withComponents(components: any[]): any {
```
Comment (ppenalver) [inline src/st-breadcrumbs/st-breadcrumbs.module.ts]: Why use this?, its only for dynamic components. It's necesary here?

If not it's necesary remember remove imports, entryComponents

File: src/st-breadcrumbs/st-breadcrumbs.scss [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+         padding: 4px 5px;
+
+         &:hover {
+            color: $text-color---hover;
+            background-color: $bg-color---hover;
+         }
+
+      }
+
+      &---arrow {
```
Comment (ppenalver) [inline src/st-breadcrumbs/st-breadcrumbs.scss]: try to not use vars and class anidation and use a css instead a scss file.

File: src/st-breadcrumbs/st-breadcrumbs.component.ts (hunk 1 above)
Comment (calcerrada) [inline src/st-breadcrumbs/st-breadcrumbs.component.ts]: The slice() method returns a shallow copy of a portion of an array into a new array

Comment (calcerrada) [inline src/st-breadcrumbs/st-breadcrumbs.component.ts]: The slice() method returns a shallow copy of a portion of an array into a new array

**Comment by ppenalver** [comment]
You need to put what issue close

**Pull request closed without merging**